apps/sms/js/thread_ui.js

```diff
@@ -1470,13 +1473,17 @@ var ThreadUI = {
 
     var onRenderMessage = (function renderMessage(message) {
       if (this._stopRenderingNextStep) {
-        // stop the iteration
+        // stop the iteration and clear the taskQueue
+        taskQueue = null;
         return false;
       }
-      this.appendMessage(message,/*hidden*/ true);
+      taskQueue.push(() => this._stopRenderingNextStep ?
```

Comment:
maybe if we use plain if it would be more readable, what do you think?
```
taskQueue.push(() => {
  if (!this._stopRenderingNextStep) {
    this.appendMessage(message,/*hidden*/ true);
  }
});
```